microsoft-edge/dualengine/reference/idualengine20browsersession.md

```diff
@@ -0,0 +1,298 @@
+
+> public HRESULT [SetHostIntranetInInternetExplorer](#sethostintranetininternetexplorer)(BOOL fEnable)
+
+#### Parameters
+* `fEnable` If true navigation to Intranet sites will stay in the Session, otherwise whether it will stay in Session or not follows default determination logic.
+
+#### SetHybridMode
+
+Sets whether to enable Hybrid Mode site list handling.
+
+> public HRESULT [SetHybridMode](#sethybridmode)(BOOL fEnable)
+
+#### Parameters
+* `fEnable` If true, Hybrid Mode behavior is enabled when handling site list urls, otherwise normal navigation behavior occurs.
+
+#### SetReverseCookieSync
+
+Sets whether cookie changes are synced back to the host.
+
+> public HRESULT [SetReverseCookieSync](#setreversecookiesync)(BOOL fEnable)
+
+#### Parameters
+* `fEnable` If true, the host will be notified of cookie changes by the IDualEngine20BrowserSessionObserver::ReverseSyncCookies() event otherwise it is not.
+
+#### SetSitePermissions
+
+Sets site permissions for the Session.
+
+> public HRESULT [SetSitePermissions](#setsitepermissions)(SITE_PERMISSION_KIND sitePermissionKind, LPCWSTR pszPermissionData)
+
+#### Parameters
+* `sitePermissionKind` The type of site permission that is being set. 
+
+* `pszPermissionData` A serialized string of the permission data to set.
+
+`pszPermissionData` is a string in the following BNF format: 
+```
+permission-data     = default-action CRLF site-permissions
+
+; Indicates the default action to take if there is no specific rule for the URI.
+; DefaultAllow - Allow permission requests.
+; DefaultAsk - Prompt the user for how they want to respond to the request.
+; DefaultBlock - Block permission requests.
+default-action      = "DefaultAllow" | "DefaultAsk" | "DefaultBlock"
+
+; A list of sites and the action to take when permission is requested.
+site-permissions    = *(permission SP wildcard SP scheme-part SP host-part SP
+                        port-part SP path-part CRLF)
+
+; Indicates the action to take on a request.
+; A - Allow permission requests.
+; Q - Prompt the user for how they want to respond to the request.
+; B - Block permission requests.
+permission          = "A" | "Q" | "B"
+
+; Indicates if the host contains a domain wildcard.
+wildcard            = BIT
+
+; Can be either a specific scheme, a wildcard to match all schemes, or "!" to indicate no scheme.
+scheme-part         = "*" | "!" | scheme
+
+; Can be either a specific hostname or a wildcard to match all hostnames.
+host-part           = "*" | host
+
+; Represents a mostly typical URI host.
+; The exception is that the host may start with a domain wildcard (e.g [*.]) which matches the host
+; and any subdomains.
+host                = ([ subdomain-wildcard ] reg-name) | IPv4address | IP-literal
+subdomain-wildcard  = "[*.]"
+
+; Can be either a specific path or a wildcard to match all paths.
+path-part           = "*" | path
+
+; Can be either a specific port or a wildcard to match all port.

```

Comment:
```suggestion
; Can be either a specific port or a wildcard to match all ports.
```

Reply:
Can be either a specific port or a wildcard to match all ports.
@AndrewLakeMSFT - add 's' in source file; from "all port" to "all ports".

Reply:
fresh Files Changed view shows the new text, "ports":
; Can be either a specific port or a wildcard to match all ports.